### PR TITLE
Schedules import

### PR DESCRIPTION
**Pull request by iamsitting**
Schedules import (#238) is done with schedules.utils.split_schedule.

It seems like it does what we want it to do:

- Mid Term
  - Splits schedule at current week
  - Assigns trainees to new schedule
  - Clears trainees from old schedule
  - Split schedule won't be imported to next term

- New Term
  - New term is created
  - Schedules are imported to next term
  - Clears trainees from imported schedule

Not sure what else to test for?

### CODE REVIEW

File: ap/schedules/models.py [hunk 1]
```diff
@@ -380,10 +377,9 @@ def assign_trainees_to_schedule(self):
     else:
       week = term.term_week_of_date(datetime.today().date())
 
-    weeks = eval(self.weeks)
+    weeks_set = Set(eval(self.weeks))
```
Comment (billyeh) [inline ap/schedules/models.py]: This is now a built-in data type: just use `set(eval(self.weeks))`. No need for `from sets import Set` either.

File: ap/apimport/utils.py [hunk 2]
```diff
@@ -690,7 +687,7 @@ def migrate_schedules():
   schedule_set.extend(schedules)
 
   for schedule in schedule_set:
-    s_new = migrate_schedule(schedule)
+    migrate_schedule(schedule)
```
Comment (jontey) [inline ap/apimport/utils.py]: migrate_schedules function should be smarter to determine if import is run during midterm or not. 

if it is not midterm, then it should copy all schedules from previous terms. Also it has to check if that schedule has already been imported. (Sometimes we run import function multiple times during the first week).

if it is midterm, (this means that we are running import after teams have been finalized, aka after draft night). then we need to run split_schedule ONLY if the trainee set gets changed, meaning new trainees have been added/removed from the team. 

i believe that this is what's not implemented yet.

**Comment by ruthnan** [comment]
It works if we merge hc-forms, because I have a fix in there for the permissions....

**Comment by iamsitting** [comment]
@yipingbenjaminliu If you want to look at this it would be good. There are a few items that need to be addressed or tested:
- [ ] MapZen to Google Api #722 
- [ ] Schedules import

File: ap/schedules/models.py [hunk 3]
```diff
@@ -359,6 +359,41 @@ def get_roll_table_by_type_in_weeks(trainees, type, weeks, team=None):
     w_tb = EventUtils.collapse_priority_event_trainee_table(weeks, schedules, t_set)
     return EventUtils.flip_roll_list(EventUtils.export_typed_ordered_roll_list(w_tb, type))
 
+  def save(self, *args, **kwargs):
+    self.save_query_filter()
+    super(Schedule, self).save(*args, **kwargs)
+
+  def save_query_filter(self):
```
Comment (iamsitting) [inline ap/schedules/models.py]: I'm not sure if we still need this...

**Pull request merged**